src/lib/aws.ts

```diff
@@ -155,13 +156,15 @@ export async function getTaskResourcesByRunId(
 
     console.log(`AWS: START: Getting a run with run id ${runId} ...`)
     const result = await client.send(command)
-    console.log('AWS:   END: GetResourcesCommand finished')
-    return result
+    result.ResourceTagMappingList = ensureValueWithError(result.ResourceTagMappingList)
+    result.PaginationToken = ensureValueWithError(result.PaginationToken)
```

Comment:
I'm curious what we end up doing with this pagination token?

Reply:
I expect handling pagination for [SHRMP-55](https://openstax.atlassian.net/browse/SHRMP-55) will use this value. For now, the reason to require that it exists is mainly to detect errors in the AWS API response. AWS seems to return `''` rather than `undefined` if there is only one page, so if this value is `undefined`, it indicates the response has gone wrong or is incomplete

Reply:
:thinking: - Perhaps this is for future consideration then, but I believe if we use an idiom like [this](https://community.aws/content/2bSFRDjmNgZUHMik3v9TZvx0rHT/how-to-use-paginator-functions-in-the-aws-sdk-for-javascript-v3), we can avoid having to manually look at the token which may make life easier :man_shrugging: . Specifically, in this case I think we can use `paginateGetResources`.